CONTRIBUTING.md

```diff
@@ -379,20 +379,36 @@ or
 
 And similarly for other scripts.
 
-## Automation of image building
+## Docker images
 
-When you run tests or enter environment (see below) the first time you do it, the local image will be
-pulled and build for you automatically.
 
-Note that building image first time pulls the pre-built version of image from Dockerhub based on master 
-sources and rebuilds the layers that need to be rebuilt - because they changed in local sources. 
-This might take a bit of time when you run it for the first time and when you add new dependencies - 
-but rebuilding the image should be an operation done quite rarely (mostly when you start seeing some
-unknown problems and want to refresh the environment). 
+For all development related tasks related to integration tests and static code checks we are using Docker
+images that are maintained in Dockerhub under `apache/airflow` repository.
+
+There are two images that we currently manage:
+
+* Slim CI image that is used for static code checks (size around 500MB) - labelled following the pattern
+  of <BRANCH>-python<PYTHON_VERSION>-ci-slim (for example master-python3.6-ci-slim)
+* Full CI image that is used for testing - containing a lot more test-related installed software
+  (size around 1GB)  - labelled following the pattern of <BRANCH>-python<PYTHON_VERSION>-ci
+  (for example master-python3.6-ci)
+
+When you run tests or enter environment or run local static checks, the first time you do it,
+the necessary local images will be pulled and built for you automatically from DockerHub. Then
+the scripts will check automatically if the image needs to be re-built if needed and will do that
+automatically for you.
+
+Note that building image first time pulls the pre-built version of images from Dockerhub might take a bit
+of time - but this is not happening when you only change sources. If you change some sensitive files
+like setup.py or Dockerfile, then the image will be rebuilt and it will take again some time (but
```

Comment:
"However, changes to sensitive files like `setup.py` or `Dockerfile` will trigger a rebuild that might take more time"